From 73c2a5dd42c4b7eb234bb8d609199d8cd2616a03 Mon Sep 17 00:00:00 2001
From: Lily Zhang <r58066@freescale.com>
Date: Mon, 12 Mar 2012 20:58:54 +0800
Subject: [PATCH 1180/2463] ENGR00176812-10 mx6solo sabreauto: change display
 for single IPU

mx6solo only supports single IPU, up to 2 display
by default. So (ARRAY_SIZE(sabr_fb_data) + 1 )/ 2
fb devices are registered. The board configuration
is:
HDMI: ipu-0, di-1
ldb: ipu-0, di-0, sec_ipu-0, sec_di-1, LDB_SEP0

Signed-off-by: Wayne Zou <b36644@freescale.com>
Signed-off-by: Lily Zhang <r58066@freescale.com>
---
 arch/arm/mach-mx6/board-mx6q_sabreauto.c |   17 ++++++++++-------
 1 files changed, 10 insertions(+), 7 deletions(-)

diff --git a/arch/arm/mach-mx6/board-mx6q_sabreauto.c b/arch/arm/mach-mx6/board-mx6q_sabreauto.c
index e532473..fe7431f 100644
--- a/arch/arm/mach-mx6/board-mx6q_sabreauto.c
+++ b/arch/arm/mach-mx6/board-mx6q_sabreauto.c
@@ -803,17 +803,17 @@ static struct ipuv3_fb_platform_data sabr_fb_data[] = {
 		.default_bpp		= 16,
 		.int_clk		= false,
 	}, {
-		.disp_dev		= "lcd",
-		.interface_pix_fmt	= IPU_PIX_FMT_RGB565,
-		.mode_str		= "CLAA-WVGA",
-		.default_bpp		= 16,
-		.int_clk		= false,
-	}, {
 		.disp_dev		= "ldb",
 		.interface_pix_fmt	= IPU_PIX_FMT_RGB666,
 		.mode_str		= "LDB-XGA",
 		.default_bpp		= 16,
 		.int_clk		= false,
+	}, {
+		.disp_dev               = "lcd",
+		.interface_pix_fmt      = IPU_PIX_FMT_RGB565,
+		.mode_str               = "CLAA-WVGA",
+		.default_bpp            = 16,
+		.int_clk                = false,
 	},
 };
 
@@ -1309,7 +1309,10 @@ static void __init mx6_board_init(void)
 		mipi_dsi_pdata.disp_id = 1;
 		ldb_data.ipu_id = 0;
 		ldb_data.disp_id = 0;
-		for (i = 0; i < ARRAY_SIZE(sabr_fb_data) / 2; i++)
+		ldb_data.sec_ipu_id = 0;
+		ldb_data.sec_disp_id = 1;
+		hdmi_core_data.disp_id = 1;
+		for (i = 0; i < (ARRAY_SIZE(sabr_fb_data) + 1) / 2; i++)
 			imx6q_add_ipuv3fb(i, &sabr_fb_data[i]);
 	} else
 		for (i = 0; i < ARRAY_SIZE(sabr_fb_data); i++)
-- 
1.7.7.4

